tests/test_pred_succ.py

```diff
@@ -124,4 +124,68 @@ def test_bfs_succesors(self):
         }
         self.assertEqual(expected, res)
         self.assertEqual([(7, [8]), (8, [9]), (9, [10])],
-                         retworkx.bfs_successors(dag, node_h))
+                         list(retworkx.bfs_successors(dag, node_h)))
+
+    def test_bfs_successors_sequence(self):
+        dag = retworkx.PyDAG()
+        node_a = dag.add_node(0)
+        node_b = dag.add_child(node_a, 1, {})
+        node_c = dag.add_child(node_b, 2, {})
+        node_d = dag.add_child(node_c, 3, {})
+        node_e = dag.add_child(node_d, 4, {})
+        node_f = dag.add_child(node_e, 5, {})
+        dag.add_child(node_f, 6, {})
+        node_h = dag.add_child(node_c, 7, {})
+        node_i = dag.add_child(node_h, 8, {})
+        node_j = dag.add_child(node_i, 9, {})
+        dag.add_child(node_j, 10, {})
+        res = retworkx.bfs_successors(dag, node_b)
+        expected = [
+            (1, [2]),
+            (2, [7, 3]),
+            (7, [8]),
+            (3, [4]),
+            (8, [9]),
+            (4, [5]),
+            (9, [10]),
+            (5, [6])
+        ]
+        for index, expected_value in enumerate(expected):
+            self.assertEqual((res[index][0], res[index][1]),
+                             expected_value)
+
+    def test_bfs_successors_sequence_invalid_index(self):
+        dag = retworkx.PyDAG()
+        node_a = dag.add_node(0)
+        node_b = dag.add_child(node_a, 1, {})
+        node_c = dag.add_child(node_b, 2, {})
+        node_d = dag.add_child(node_c, 3, {})
+        node_e = dag.add_child(node_d, 4, {})
+        node_f = dag.add_child(node_e, 5, {})
+        dag.add_child(node_f, 6, {})
+        node_h = dag.add_child(node_c, 7, {})
+        node_i = dag.add_child(node_h, 8, {})
+        node_j = dag.add_child(node_i, 9, {})
+        dag.add_child(node_j, 10, {})
+        res = retworkx.bfs_successors(dag, node_b)
+        with self.assertRaises(IndexError):
```

Comment:
I was expecting this to be a `TypeError` (for trying to index into an iterator) until I read the API docs for the `BFSSuccessors` return type.